### DISCUSSION TITLE
Decouple per_hour from hour increments

### DISCUSSION THREAD

**Discussion by zanebclark** [openmobilityfoundation/curb-data-specification#54] [Data Specification]
## Define `per_hour` to support different units of time

In the current [Rate spec](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/curbs#rate), the "rate" part of the `Rate` object (`per_hour`) is defined as an integer. If a municipality were to define rates by day, the conversion of such a rate to an hourly rate would yield a float (e.g. $100 / day = $4.16666666667 / hour). For measurements smaller than an hour, there's no potential that scaling the rate would result in a float. A per-minute integer rate (e.g. 6 cents per minute) converted to a per-hour rate would continue to be an integer (360 cents per hour), as multiplying an integer by an integer yields an integer. The same logic applies to second and sub-second rates. Still, an additional step would be required to convert CDS-served policies (360 cents per hour) back to the original rate definition (6 cents per minute). 

Proposal: Define the "rate" part of the `Rate` object with `rate` and `rate_unit` instead of `per_hour`. 

|       Name        |    Type     | Required/Optional |                                                                                  Description                                                                                  |
|:-----------------:|:-----------:|:-----------------:|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
|   ~~per_hour~~    | ~~Integer~~ |   ~~Required~~    |                                        ~~The rate per hour for this space in cents (or the smallest denomination of local currency).~~                                        |
|       rate        |   Integer   |     Required      |                                       The rate for this space in cents (or the smallest denomination of local currency) per rate_unit.                                        |
|     rate_unit     |   String    |     Required      |                                             The unit of time associated with the rate. One of "second", "minute", "hour", "day".                                              |
| increment_minutes |   Integer   |     Optional      |             If specified, this is the smallest amount of time a user can pay for (e.g., if increment_minutes is 15, a user can pay for 15, 30, 45, etc. minutes).             |
| increment_amount  |   Integer   |     Optional      |                       If specified, the rate for this space is rounded up to the nearest increment of this amount, specified in the same units as rate.                       |
|   start_minutes   |   Integer   |     Optional      | The amount of time the vehicle must have already been present in the Curb Zone before this rate starts applying. If not specified, this rate starts when the vehicle arrives. |
|    end_minutes    |   Integer   |     Optional      |                              The amount of time after which the rate stops applying. If not specified, this rate ends when the vehicle departs.                               |

Some example Rates:

2 cents per second
```json
{
  "rate": 2,
  "rate_unit": "second"
}
```

$100 per day
```json
{
  "rate": 10000,
  "rate_unit": "day"
}
```

30 cents per 15 minute interval

```json
{
  "rate": 2,
  "rate_unit": "minute",
  "increment_minutes": 15
}
```

**Comment by schnuerle** [comment]
Agree with this change. And it's like we do it in [Rules in MDS](https://github.com/openmobilityfoundation/mobility-data-specification/tree/main/policy#rules). Updated to your suggestion [here](https://github.com/openmobilityfoundation/curb-data-specification/blob/feature-release-work-1/curbs/README.md#rate).

**Reply by zanebclark** [reply]
Splendid. Thanks!
